contracts/dapis/proxies/ProxyFactory.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./DataFeedProxy.sol";
+import "./DapiProxy.sol";
+import "./DataFeedProxyWithOev.sol";
+import "./DapiProxyWithOev.sol";
+
+contract ProxyFactory {
+    event DeployedDataFeedProxy(
+        address indexed proxyAddress,
+        bytes32 indexed dataFeedId,
+        bytes metadata
+    );
+
+    event DeployedDapiProxy(
+        address indexed proxyAddress,
+        bytes32 indexed dapiName,
+        bytes metadata
+    );
+
+    event DeployedDataFeedProxyWithOev(
+        address indexed proxyAddress,
+        bytes32 indexed dataFeedId,
+        address indexed oevBeneficiary,
+        bytes metadata
+    );
+
+    event DeployedDapiProxyWithOev(
+        address indexed proxyAddress,
+        bytes32 indexed dapiName,
+        address indexed oevBeneficiary,
+        bytes metadata
+    );
+
+    address public immutable dapiServer;
+
+    constructor(address _dapiServer) {
+        require(_dapiServer != address(0), "dAPI server zero");
+        dapiServer = _dapiServer;
+    }
```

Comment:
Note that `dapiServer` can also be provided as an argument to the deployment functions, but I figured this is less error-prone